modules/search/class.jetpack-search.php

```diff
@@ -370,7 +370,12 @@ public function search( array $es_args ) {
 		}
 
 		$response_code = wp_remote_retrieve_response_code( $request );
-		$response      = json_decode( wp_remote_retrieve_body( $request ), true );
+
+		if ( ! $response_code || $response_code < 200 || $response_code >= 300 ) {
```

Comment:
This check has been moved down in JP: https://github.com/Automattic/jetpack/blob/f0678c20f4e2c9b6cb5baacda15d0014921fab70/modules/search/class.jetpack-search.php#L414-L430